docs/core/testing/unit-testing-mstest-migration-from-v1-to-v3.md

```diff
@@ -0,0 +1,290 @@
+
+---
+
+### Who Is Impacted?
+
+This guide is intended for projects currently using MSTest v1 via either:
+
+- **Assembly references**: Projects referencing `Microsoft.VisualStudio.QualityTools.UnitTestFramework.dll` directly in their project files.
+- **NuGet packages**: Projects using MSTest v1 through the `MSTest.TestFramework` NuGet package.
+
+If your project relies on MSTest for unit testing and includes the above references, it will benefit from the improvements in MSTest v3 and requires adjustments outlined in this guide.
+
+---
+
+### Why Migrate to MSTest v3?
+
+Even if you’re satisfied with your current MSTest setup, upgrading to MSTest v3 unlocks substantial advantages that improve both the quality and future-readiness of your tests. Here’s why making the switch now can be a valuable step forward:
+
+- **Immediate Performance Boosts**: MSTest v3 significantly reduces test execution time and optimizes resource usage. This is particularly beneficial in CI/CD pipelines, where faster tests can shorten deployment cycles and cut infrastructure costs.
+
+- **Future-Proofing Your Tests**: MSTest v3 offers robust support for modern .NET versions, including .NET 6, .NET 7, and future iterations, along with cross-platform compatibility. This means your tests will be more adaptable and ready for upgrades, avoiding the technical debt of outdated testing frameworks.
+
+- **Increased Code Reliability with New Analyzers**: MSTest v3 built-in code analyzers enforce best practices, helping to catch issues early and promoting cleaner, more maintainable test code:
+  - **Proactive Issue Detection**: The analyzers provide real-time feedback, suggesting improvements and flagging potential issues as you write your tests.
+  - **Stronger Type-Safe Assertions**: Replacing ambiguous overloads, MSTest v3 enforces type-safe assertions, reducing the risk of incorrect test behavior and making tests more reliable.
+  - **Cleaner Code, Less Maintenance**: By aligning your tests with MSTest standards and best practices, MSTest v3 helps you maintain a cleaner, more manageable codebase, reducing technical debt over time.
+
+- **Greater Flexibility and Extensibility**: MSTest v3 supports advanced testing scenarios, including dynamic data sources and in-assembly parallel execution. This flexibility enables more sophisticated testing approaches and speeds up test suites without complex configurations.
+
+By upgrading, you’re setting up your tests to be faster, more reliable, and adaptable to future .NET developments, positioning your project for long-term success and easier maintenance.
+
+---
+
+## Steps to Migrate
+
+### 1. Remove Assembly Reference
+
+For projects using MSTest v1 through assembly references, it’s common to find references to either or both of the following DLLs:
+
+- `Microsoft.VisualStudio.QualityTools.UnitTestFramework`
+- `Microsoft.VisualStudio.TestPlatform.TestFramework`
```

Comment:
Where is this name coming from? I don't think that is used by legacy projects, 
![image](https://github.com/user-attachments/assets/a11997a9-a592-4c36-879f-f6caa592882d)
The .common is imho the one that is referenced, and they are always referenced both.